File: TransCoder_tokenized_test_set_functions/cpp/REORDER_A_ARRAY_ACCORDING_TO_GIVEN_INDEXES_1.cpp
void reorder ( int arr [ ], int index [ ], int n ) {
  for ( int i = 0;
  i < n;
  i ++ ) {
    while ( index [ i ] != i ) {
      int oldTargetI = index [ index [ i ] ];
      char oldTargetE = arr [ index [ i ] ];
      arr [ index [ i ] ] = arr [ i ];
      index [ index [ i ] ] = index [ i ];
      index [ i ] = oldTargetI;
      arr [ i ] = oldTargetE;
    }
  }
}

File: TransCoder_tokenized_test_set_functions/python/REORDER_A_ARRAY_ACCORDING_TO_GIVEN_INDEXES_1.py
def reorder ( arr, index, n ) :
    for i in range ( 0, n ) :
        while ( index [ i ] != i ) :
            oldTargetI = index [ index [ i ] ]
            oldTargetE = arr [ index [ i ] ]
            arr [ index [ i ] ] = arr [ i ]
            index [ index [ i ] ] = index [ i ]
            index [ i ] = oldTargetI
            arr [ i ] = oldTargetE
    
